Documentation/contributing/testing/scalability.rst

```diff
@@ -0,0 +1,294 @@
+* ``generatedConfig_netpol.yaml`` - Rendered test scenario
+* ``'GenericPrometheusQuery NetPol Average CPU Usage_netpol_.*.json'`` - ``GenericPrometheusQuery`` test results contain results of the prometheus queries executed during test.
+  In this example, it contains CPU usage of the Cilium agents. All of the Prometheus Queries will be automatically
+  visualized in :ref:`perfdash <perfdashdocs>`.
+* ``'PodPeriodicCommand.*Profiles-stdout.*'`` - Contains memory and CPU profiles gathered during the test run. To see how to read them, please refer to the :ref:`profiling`
+
+
+Accessing Grafana and Prometheus during test run
+""""""""""""""""""""""""""""""""""""""""""""""""
+
+During test execution, ClusterLoader2 deploys Prometheus and Grafana to the cluster.
+You can access Grafana and Prometheus by running:
+
+.. code-block:: bash
+
+    kubectl port-forward -n monitoring svc/grafana 3000
+    kubectl port-forward -n monitoring svc/prometheus-k8s 9090
+
+This can be especially useful for exploring the metrics and adding additional queries to the test.
+
+Metrics-based testing and alerting
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Sometimes, you might want to scrape additional targets during test execution on top of the default ones.
+In this case, you can simply create a Pod or Service monitor `example monitor <EXAMPLE_MONITOR_>`_.
+Then you need to pass it as additional argument to ClusterLoader2:
+
+.. code-block:: bash
+
+    ./clusterloader \
+    --prometheus-additional-monitors-path=../../.github/actions/cl2-modules/egw/prom-extra-podmons
+    ...
+
+Now you can use the additional metrics in your test, by leveraging regular ``GenericPrometheusQuery`` measurement.
+For example, Egress Gateway ensures that various percentiles of masquerade latency observed by client are
+`below specific thresholds <EGW_MASQ_METRICS_>`_. This can be achieved by following measurement in ClusterLoader2:
+
+.. code-block:: yaml
+
+  - Identifier: MasqueradeDelay{{ .metricsSuffix }}
+    Method: GenericPrometheusQuery
+    Params:
+      action: {{ .action }}
+      metricName: Masquerade Delay {{ .metricsSuffix }}
+      metricVersion: v1
+      unit: s
+      enableViolations: true
+      queries:
+      - name: P95
+        query: quantile(0.95, egw_scale_test_masquerade_delay_seconds_total{k8s_instance="{{ .instance }}"})
+        threshold: {{ $MASQ_DELAY_THRESHOLD }}
+
+
+Running tests in CI
+~~~~~~~~~~~~~~~~~~~
+
+Once you are happy with the test and validated it locally, you can create a PR with the test.
+You can base your Github workflow on the existing tests, or add test scenario to already existing workflow.
+
+
+Accessing test results from PR or CI runs
+"""""""""""""""""""""""""""""""""""""""""
+
+You can run the specific scalability or performance test in your PR, some example commands are:
+
+.. code-block:: bash
+
+    /scale-100
+    /scale-clustermesh
+    /scale-egw
+    /fqdn-perf
+
+After test run, all of the results will be saved in GS bucket.
```

Comment:
```suggestion
After the test run, all of the results will be saved in GS bucket.
```

Please expand GS - do you mean Google Cloud Storage?

Reply:
yes, done.